### PR TITLE
Make `yarn postinstall` cross-env

### PR DESCRIPTION
**Pull request by sindrig**
It should work on windows without the linux subsystem

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "help": "nx help",
     "ts-node": "ts-node -r tsconfig-paths/register -O '{\"module\":\"commonjs\",\"esModuleInterop\":true}'",
     "schemas": "test -f libs/api/schema/src/lib/schema.d.ts || echo 'export default () => {}' >> libs/api/schema/src/lib/schema.d.ts && yarn nx run api:contentful-types && yarn nx run application-system-api:build-schema && yarn nx run api-domains-application:codegen && yarn nx run api:build-schema && yarn affected:schemas",
-    "postinstall": "if [ ! -z ${CI+x} ]; then exit 0; else yarn schemas; fi"
+    "postinstall": "ts-node ./scripts/postinstall.js"
```
Comment (darrikonn) [inline package.json:39]: just `node` since the script is a js file.

**Review by darrikonn — APPROVED** [review]
One comment, otherwise approved.

**Review by dabbeg — APPROVED** [review]

**Review by jeremybarbet — DISMISSED** [review]
LGTM

**Review by jeremybarbet — DISMISSED** [review]
LVGTM

**Comment by sindrig** [comment]
Sorry about the noise in here - this got a little bit bigger than anticipated :)

File: scripts/Dockerfile [hunk 2]
```diff
@@ -7,6 +7,9 @@ WORKDIR /build
 
 # Adding and installing packages
 ADD package.json yarn.lock ./
+
+RUN mkdir -p /build/scripts
+ADD ./scripts/postinstall.js /build/scripts/
```
Comment (jeremybarbet) [inline scripts/Dockerfile:12]: That's what I was missing on docker when I tried to export my script out of package.json. Good to know 👍

**Review by jeremybarbet — APPROVED** [review]

File: scripts/Dockerfile (hunk 2 above)
Comment (sindrig) [reply scripts/Dockerfile:12]: Long term I'd like scripts to only contain things that pertain to developers/building and move all ci stuff to a separate folder (or maybe to `scripts/ci`) idk. anyway.

**Comment by jeremybarbet** [comment]
@sindrig It's failing to run the prepare step successfully

```
Running yarn affected:schemas
Error: Command failed: yarn affected:schemas
fatal: Not a valid object name master
nx affected

Run task for affected projects

[...]

Error: Command failed: git merge-base master HEAD
fatal: Not a valid object name master
```

File: scripts/generate-schemas.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  'yarn nx run api-domains-application:codegen',
+  'yarn nx run api:build-schema',
+  'yarn affected:schemas',
+]
+
+const main = async () => {
+  if (!(await promisify(fs.exists)(SCHEMA_PATH))) {
+    await promisify(fs.writeFile)(SCHEMA_PATH, 'export default () => {}')
+  }
+  for (cmd of YARN_COMMANDS) {
```
Comment (darrikonn) [inline scripts/generate-schemas.js:18]: Better to use `forEach/map/reduce` (in this case, `forEach`). These looping constructs treat empty elements differently.

Comment (sindrig) [reply scripts/generate-schemas.js]: After DM we decided not to do that

**Review by jeremybarbet — APPROVED** [review]

**Review by dabbeg — APPROVED** [review]

**Review by maggisk — APPROVED** [review]

**Pull request merged**